tsk/fs/ext2fs.c

```diff
@@ -1174,7 +1178,11 @@ ext2fs_inode_walk(TSK_FS_INFO * fs, TSK_INUM_T start_inum,
     size =
         ext2fs->inode_size >
         sizeof(ext2fs_inode) ? ext2fs->inode_size : sizeof(ext2fs_inode);
-    if ((dino_buf = (ext2fs_inode *) tsk_malloc(size)) == NULL) {
+
+    dino_buf = (ext2fs_inode *) tsk_malloc(size);
+    if (dino_buf == NULL) {
+        free(fs_file->meta);
```

Comment:
Hi @joachimmetz .  Does calling tsk_fs_file_close() not work here instead of the two calls to free fs_file and fs_file->meta?

Reply:
maybe, I find the current naming/calling convention confusing, alloc to "open" the fs_file and "close" to free it? 

if that is the case I would recommend creating tsk_fs_file_open()

I would also recommend using `goto on_error` for a more clear and cleaner error path

Reply:
for now I've mainly addressed the leak itself